Add CI

HumzahJavid opened the pull request:
Closes #5 and #2

## Description
This PR will:
- Add a dependabot.yml
- Add a Build check
- Add a Linting check (via prettier)
- Add cypress tests and check (with upload of screenshots on failure)

keiranjprice101 commented on cypress/e2e/cypress_spec.cy.ts:
```diff
@@ -0,0 +1,16 @@
+// This file tests the cypress library is working correctly
```
I don't think we need this. Can we remove this file?

HumzahJavid replied:
I thought it could be useful in helping to diagnose failing tests where the cause could be the testing library itself.

But yes, it can be removed.

keiranjprice101 commented on cypress/e2e/spec.cy.ts:
```diff
@@ -0,0 +1,137 @@
+describe('E2E Test', () => {
+  it('Opens the root page', () => {
+    cy.visit('http://localhost:3000');
```
Can you look at adding a `beforeEach` block and moving all the `cy.visit` calls into that single block?

HumzahJavid replied:
I've added a beforeEach block and moved the first `cy.visit('http://localhost:3000');` in most test cases, save for the one which needs to pre-register a GET and PUT.

keiranjprice101 commented on cypress/e2e/spec.cy.ts:
```diff
@@ -0,0 +1,137 @@
+});
+
+describe('E2E Test', () => {
+  // const exampleJSON = require('../fixtures/example.json');
```
Do we need this comment?

keiranjprice101 commented on cypress/e2e/spec.cy.ts:
```diff
@@ -0,0 +1,137 @@
+    });
+
+    // works but no assert
+    // cy.get('[data-cy="ButtonAlert"]').contains('Specification update failed');
```
Do we need this commented line?

keiranjprice101 reviewed (changes requested):
Looks good and was working when I ran the tests locally. 

Just a few comments about leftover code in comments that could probably be removed.

Could you also add something to the readme about how to run cypress locally

Just something simple like "To run the cypress suite locally: `yarn cypress open`

HumzahJavid commented:
I've added a beforeEach block and moved the first cy.visit('http://localhost:3000'); in most test cases, save for the one which needs to pre-register a GET and PUT.

And removed all the marked comments.
